File: backend/app/integrations/flashcards/structures.py
from typing import List, Optional
from pydantic import BaseModel, Field

class FlashCard(BaseModel):
    topic: str = Field(description="Main topic or subject this flashcard belongs to.")
    subtopic: Optional[str] = Field(None, description="Optional subtopic or section of the document.")
    
    question: str = Field(description="Question or prompt for the flashcard.")
    answer: str = Field(description="Main answer text for the flashcard.")
    
    key_terms: List[str] = Field(default_factory=list, description="Important terms or keywords relevant to the question.")
    difficulty: Optional[str] = Field(description="Difficulty level: easy, medium, or hard.")
    
    explanation: Optional[str] = Field(description="Expanded explanation or context behind the answer.")
        
    tags: List[str] = Field(default_factory=list, description="Tags or categories to help organize the flashcard.")

class FlashCardSet(BaseModel):
    flashcards: List[FlashCard] = Field(description="List of flashcards in this set.")

File: frontend/src/nano/exercises/Structures.ts
export const multipleChoiceStructure = `[
   {
     "topic": "string (optional, e.g. 'Mathematics')",
     "difficulty": "string (optional, one of: Easy, Medium, Hard)",
     "question": "string (the exercise question)",
     "choices": [
       { "label": "A", "text": "string", "is_correct": true/false },
       { "label": "B", "text": "string", "is_correct": true/false }
     ],
     "explanation": "string (optional, why the answer is correct)",
     "learning_objective": "string (optional, the learning objective tested)"
   }
 ]`